SL/Mailer.pm

```diff
@@ -22,143 +31,122 @@ sub new {
   bless $self, $type;
 }
 
-
 sub send {
-  my ($self, $out) = @_;
-
-  my $boundary = time;
-  my $domain = $self->{from};
-  $domain =~ s/(.*?\@|>)//g;
-  my $msgid = "$boundary\@$domain";
-  $boundary = "SL-$self->{version}-$boundary";
-  
-  $self->{charset} ||= "ISO-8859-1";
-  
-  if ($out) {
-    open(OUT, $out) or return "$out : $!";
-  } else {
-    open(OUT, ">-") or return "STDOUT : $!";
-  }
+  my ($self) = @_;
+
+  #----- CONFIGURATION
+  my $tmpfolder = 'http://domain.net/ledger123/rma/users';
+  my $apikey = 'xkeysib-ad3c502c031275016ba77259cbaad630f58431abfa6eb4938c56e62e7dc64591-UfY14WhVRGN';
+  #----- CONFIG END
+
+  $self->{contenttype} = "text/plain" unless $self->{contenttype};
 
-  $self->{contenttype} ||= "text/plain";
-  
-  my %h;
-  for (qw(reply-to from to cc bcc)) {
+  for (qw(from to cc bcc)) {
     $self->{$_} =~ s/\&lt;/</g;
     $self->{$_} =~ s/\&gt;/>/g;
     $self->{$_} =~ s/(\/|\\|\$)//g;
-    $self->{$_} =~ s/["]?(.*?)["]? (<.*>)/"=?$self->{charset}?B?".&encode_base64($1,"")."?= $2"/e
-      if $self->{$_} =~ m/[\x00-\x1F]|[\x7B-\xFFFF]/;
-    $h{$_} = $self->{$_};
-  }
- 
-  $h{'reply-to'} = "Reply-to: $h{'reply-to'}\n" if $self->{'reply-to'};
-  $h{cc} = "Cc: $h{cc}\n" if $self->{cc};
-  $h{bcc} = "Bcc: $h{bcc}\n" if $self->{bcc};
-  $h{subject} = ($self->{subject} =~ /([\x00-\x1F]|[\x7B-\xFFFF])/) ? "Subject: =?$self->{charset}?B?".&encode_base64($self->{subject},"")."?=" : "Subject: $self->{subject}";
-    
-  $now_string = strftime "%a, %d %b %Y %H:%M:%S %z", localtime;
-  $h{date} = "Date: " . $now_string . "\n";
-  
-  if ($self->{notify}) {
-    if ($self->{notify} =~ /\@/) {
-      $h{notify} = "Disposition-Notification-To: $self->{notify}\n";
-    } else {
-      $h{notify} = "Disposition-Notification-To: $h{from}\n";
-    }
   }
-  
-  print OUT qq|From: $h{from}
-To: $h{to}
-$h{date}$h{'reply-to'}$h{cc}$h{bcc}$h{subject}
-Message-ID: <$msgid>
-$h{notify}X-Mailer: Run my Accounts $self->{version}
-MIME-Version: 1.0
-|;
 
+  my $json = JSON->new->utf8;
 
-  if (@{ $self->{attachments} }) {
-    print OUT qq|Content-Type: multipart/mixed; boundary="$boundary"
+  my $data = {};
 
-|;
-    if ($self->{message} ne "") {
-      print OUT qq|--${boundary}
-Content-Type: $self->{contenttype}; charset="$self->{charset}"
+  $data->{sender}->{name} = $self->{fromname};
+  $data->{sender}->{email} = $self->{from};
 
-$self->{message}
+  if ($self->{replyto}){
+     $data->{replyTo}->{name} = $self->{replyto};
```

Comment:
shouldn't it be the same as $self->{fromname}; ?

Reply:
Yes, you are right. You have sharp eye. I fix it.